packages/product/src/models/product-image.ts

```diff
@@ -6,27 +6,33 @@ import {
   Index,
   ManyToMany,
   OnInit,
-  OptionalProps,
   PrimaryKey,
   Property,
 } from "@mikro-orm/core"
 
-import { DAL } from "@medusajs/types"
-import { DALUtils, generateEntityId } from "@medusajs/utils"
+import {
+  DALUtils,
+  createPsqlIndexStatementHelper,
+  generateEntityId,
+} from "@medusajs/utils"
 import Product from "./product"
 
-type OptionalRelations = "products"
-type OptionalFields = DAL.SoftDeletableEntityDateColumns
+const imageUrlIndexName = "IDX_product_image_url"
+const imageUrlIndexStatement = createPsqlIndexStatementHelper({
+  name: imageUrlIndexName,
+  tableName: "image",
+  columns: ["url"],
+  unique: false,
+  where: "deleted_at IS NULL",
+})
 
+imageUrlIndexStatement.MikroORMIndex()
```

Comment:
comment: In general I prefer having the index on the property itself, unless it is a multi columns, but I don't have a strong opinion but since this is what we are doing everywhere I d like to know why we changed that in this pr :)

Reply:
@adrien2p I was aiming for consistency on where we define the index, regardless if it is multi-column or not. Like that, it's easier to find what indices have been defined for a model (just by looking at the top). That is also in-line with how postgres presents the indices if you do `/d` for a table, for example.